Removed warnings

PatrickColborne opened the pull request:
*There were a bunch of warnings that were really irritating me after the move to compile on Java 11 so I have fixed them.
*I also updated a bunch of other plugins and the docker image that we are building from.

The warnings were:
```
[WARNING] The POM for com.sun.xml.bind:jaxb-core:jar:2.2.11 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
[WARNING] The POM for com.sun.xml.bind:jaxb-impl:jar:2.2.11 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
```
and they are generated by the gateway plugins that implement Java 11 compiled service-interfaces.

TimElectrum commented on pom.xml:
```diff
@@ -162,7 +163,7 @@
         <groupId>com.github.kongchen</groupId>
         <artifactId>swagger-maven-plugin</artifactId>
-        <version>3.1.3</version>
+        <version>3.1.8</version>
```
I presume this doesn't change the generated swagger in any significant way but please double check that

PatrickColborne replied:
I'll have a look. I also noticed that code coverage was failing so I have sorted that out too.

TimElectrum reviewed (changes requested):
Please add to the release notes. Release notes for our public APIs aren't the 'normal' ones in the project root, they're in the API documentation resources.

The documentation for the base API isn't actually exposed anywhere but updating the release notes anyway:

- Helps us to figure out what changed in this version when we come back to it later.
- Helps the next dev who bumps the base API version in a public API - they can mostly copy/paste your additions to the release notes.

Please also just make sure we don't change the swagger as a result of changing the swagger plugin version

PatrickColborne commented:
> Please add to the release notes. Release notes for our public APIs aren't the 'normal' ones in the project root, they're in the API documentation resources.
> 
> The documentation for the base API isn't actually exposed anywhere but updating the release notes anyway:
> 
> * Helps us to figure out what changed in this version when we come back to it later.
> * Helps the next dev who bumps the base API version in a public API - they can mostly copy/paste your additions to the release notes.
> 
> Please also just make sure we don't change the swagger as a result of changing the swagger plugin version

I have reverted the swagger-maven-plugin version change because it ended up removing some description in the swagger file. Otherwise, I have now added some release notes.

TimElectrum reviewed (changes requested):
I'm happy with the spirit of the changes but:

1. There are a couple of instances where we are using different versions of various libraries within a framework (e.g. `jackson-datatype` and `jaxb`). I'd prefer that we used the same version for all of them i.e. have a `jackson` version and a `jaxb` version. By mixing versions I'm worried we might break hidden dependencies within the frameworks (e.g. if a method is moved and we call the new/old method but it's not in the matching library).
2. I wonder if we should release an RC of this and get it into our test environments before actually releasing it? Again, I'm worried about breaking changes within these 3rd party libraries that aren't immediately apparent to us. We should do some more rigorous testing before releasing it.

PatrickColborne commented on pom.xml:
```diff
@@ -33,15 +33,16 @@
+    <jackson-datatype-jsr310-version>2.11.2</jackson-datatype-jsr310-version>
     <el-version>3.0.0</el-version>
-    <jaxb-version>2.2.11</jaxb-version>
+    <jaxb-version>2.3.1</jaxb-version>
+    <jaxb-core-version>2.3.0</jaxb-core-version>
```
I split these up because there is no version 2.3.1 for jaxb-core and I wanted to include the bug fix in the other jaxb libraries. I agree with you so I have changed it.

TimElectrum replied:
if there's no 2.3.1 for `jaxb-core` then our options are:

1. keep the versions split so we can use the latest of each or
2. amalgamate them as you've done and drop some of the JAX-B libraries from 2.3.1 to 2.3.0.

Of those two options, I prefer #1 because we shouldn't _downgrade_ libraries if we can help it.

PatrickColborne commented:
@TimElectrum I have updated this PR now but I am not entirely sure what benefit it adds. It seems like all the dependency changes that I was  making in this PR have already been made in a previous commit. I have fixed the versioning as you've suggested though.

TimElectrum commented on src/docs/devguide/hugo/site/content/release-notes.md:
```diff
@@ -1,5 +1,9 @@
 
+## 3.31.1
+Released 11 June 2021
+* Changed `jaxb` and `jackson` dependencies to share a version in order to instill consistency.
```
```suggestion
* Changed `jackson` dependencies to share a version in order to instill consistency.
```
